### PR TITLE
Paypal ipn options

### LINKED ISSUE #7014: Subscription not expiring at the end of a 12 month plan
### General Info
- UUID: a75eb6f8-5a29-482f-a971-7aff7438da86
- Browser: Firefox 45
- OS: Ubuntu 15.10
### Description

I paid a 12 months subscription, as far as I can remember somewhere around the end of February 2015. It should have been automatically renewed at the end of February this year. Except my credit card informations changed last year, and I didn't changed them on Paypal because I don't use it that much in the first place. So it should have failed. But in fact, my subscription didn't even expired, as far as I can see it.

I was waiting for the renewal to change my card informations. On March 8, I received a notification from Paypal stating that the automatic payment failed (logically, since informations were outdated). I ran to the Subscription page on Habitica, only to see that I now had 13 consecutive months, and that I was still a subscriber. I first thought that this was temporary. But I'm now at 14 consecutive months, still a subscriber. I was able to buy gems, got the subscribers monthly reward; the only thing missing is the hourglass, which were not renewed (maybe I would have got one on the 15th month?)

This is annoying because:
- I feel like I abused the system
- I'm willing to give my money to a project that I really like
- I can't seem to get any hourglass

Also, just to make sure, I checked my bank account and can confirm that I wasn't charged in any way.

I guess this can be solved by cancelling and then pay a new subscription, but maybe doing so wouldn't allow you to get the information you need to look into this issue. I'm waiting for any input on how to make things right.

### PR DESCRIPTION
**Pull request by TheHollidayInn**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #7014

"When PayPal can't take a payment, it tries again later which is good, but apparently it doesn't tell Habitica, so Habitica can't inform the user. For a 12-month subscription, this could mean that the user gets an entire year free. https://github.com/HabitRPG/habitrpg/issues/7014"

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
This adds more ipn notifications that count as cancelling a subscription.

### CODE REVIEW

File: test/api/v3/integration/tasks/PUT-tasks_id.test.js [hunk 1]
```diff
@@ -389,18 +390,20 @@ describe('PUT /tasks/:id', () => {
       });
     });
 
-    it('updates a daily', async () => {
+    it.only('updates a daily', async () => {
```
Comment (SabreCat) [inline test/api/v3/integration/tasks/PUT-tasks_id.test.js]: Accidental commit of an `only` (lint will probably gripe at ya)

File: website/server/libs/paypalPayments.js [hunk 2]
```diff
@@ -220,7 +220,14 @@ api.ipn = async function ipnApi (options = {}) {
 
   let {txn_type, recurring_payment_id} = options;
 
-  if (['recurring_payment_profile_cancel', 'subscr_cancel'].indexOf(txn_type) === -1) return;
+  let ipnAccetableTypes = [
```
Comment (SabreCat) [inline website/server/libs/paypalPayments.js]: Not crucial, but this is an awkward-looking typo `Accetable`

File: test/api/v3/integration/tasks/POST-tasks_user.test.js [hunk 3]
```diff
@@ -509,6 +509,7 @@ describe('POST /tasks/user', () => {
       expect(task.daysOfMonth).to.eql([15]);
       expect(task.weeksOfMonth).to.eql([3]);
       expect(new Date(task.startDate)).to.eql(now);
+      expect(task.isDue).to.be.true;
```
Comment (SabreCat) [inline test/api/v3/integration/tasks/POST-tasks_user.test.js]: Actually, what do these `isDue` changes have to do with PayPal IPN? Was this supposed to be part of a different branch?

Comment (TheHollidayInn) [reply test/api/v3/integration/tasks/POST-tasks_user.test.js]: Oops yes. My mistake

**Pull request merged**